### ISSUE TITLE
different result every inference by load pretrained model

### ISSUE THREAD

**Issue by Feywell** [NVIDIA/tacotron2#336]
Why there is different result after inference？There is a puzzle on LinearNorm Class.

**the same text with different mel_out:**

the **first:**
![image](https://user-images.githubusercontent.com/24384298/79685329-8e135f80-826a-11ea-8b4e-59ad952f7142.png)
**the second:**
![image](https://user-images.githubusercontent.com/24384298/79685344-aedbb500-826a-11ea-8319-707422687a9f.png)

```
`class LinearNorm(torch.nn.Module):
    def __init__(self, in_dim, out_dim, bias=True, w_init_gain='linear'):
        super(LinearNorm, self).__init__()
        self.linear_layer = torch.nn.Linear(in_dim, out_dim, bias=bias)

        torch.nn.init.xavier_uniform_(
            self.linear_layer.weight,
            gain=torch.nn.init.calculate_gain(w_init_gain))

    def forward(self, x):
        return self.linear_layer(x)`
```

Does The init function run every forward?
 The result is correct , when I set torch.manual_seed(SEED).
How can I solve this problem?
Thank you!

 Does anyone encounter the same question?

**Comment by EuphoriaCelestial** [comment]
I have the same problem, ran inference for 1 sentence multiple times, got different audio results.
Sometime its just differ in pitch, noise, volume, but sometimes it got echo in the end of sentence

**Comment by pravn** [comment]
I wonder if we are looking at something like ensembling or distillation for
these. Very researchy.
pravn.wordpress.com

**Comment by srijan14** [comment]
I happen to have the same problem. ran inference on my model first time, the results were great. But now when running again, the results aren't that good and not sure how can i replicate the old state.

**Comment by rafaelvalle** [comment]
This is a feature that allows the model to the say the same phrase in multiple ways, just like humans would do.
It is related to the [Dropout on the Prenet](https://github.com/NVIDIA/tacotron2/blob/master/model.py#L99).

**Comment by EuphoriaCelestial** [comment]
@rafaelvalle  what if I change p to 0? will the result be the same every time?

> x = F.dropout(F.relu(linear(x)), p=0.5, training=True)

**Comment by Feywell** [comment]
@rafaelvalle Actually, if we use model.eval(), dropout is not work. Is it same every time in your experiment？

**Comment by rafaelvalle** [comment]
The model will not work with model.eval() nor with p dropout values far away from 0.5.

**Comment by wmathor** [comment]
@rafaelvalle 
what if i change `training` to False? will the result be the same every time?
`x = F.dropout(F.relu(linear(x)), p=0.5, training=False)`

**Comment by creotiv** [comment]
> @rafaelvalle
> what if i change `training` to False? will the result be the same every time?
> `x = F.dropout(F.relu(linear(x)), p=0.5, training=False)`

Then it will generate only noice
